cl/scrapers/management/commands/scrape_caselaw.py

```diff
@@ -0,0 +1,339 @@
+    @return: reporter slug or None
+    """
+    global reporters_slugs
+    if reporters_slugs.get(reporter_endpoint):
+        # Check if slug is already stored in global variable to avoid
+        # unnecessary requests
+        return reporters_slugs.get(reporter_endpoint)
+    data = get_data(reporter_endpoint)
+    if data:
+        frontend_url = data.get("frontend_url")
+        if frontend_url:
+            url = [y for y in frontend_url.split("/") if y]
+            if url:
+                # Store and return reporter slug from url
+                reporters_slugs[reporter_endpoint] = url[-1]
+                return url[-1]
+    return None
+
+
+def get_from_caselaw(options: OptionsType):
+    """Read options, fetch endpoint and store results
+
+    :param options: dict with options passed to management command
+    Usage examples:
+
+    Get the cases from 'md' court with last updated date greater than or equal to year
+    2022, with full case text in xml
+    manage.py scrape_caselaw --type cases --full-case True --body-format xml --court md
+    --last- updated 2022 --last-updated-filter gte
+
+    Get all the courts
+    manage.py scrape_caselaw --type courts
+
+    Get all fastcase cases
+    manage.py scrape_caselaw --type cases --source Fastcase --full-case True --body-format xml
+
+    """
+    type = options["type"]
+    cursor = options["cursor"]
+    court = options["court"]
+    jurisdiction = options["jurisdiction"]
+    last_updated = options["last_updated"]
+    last_updated_filter = options["last_updated_filter"]
+    ordering = options["ordering"]
+    body_format = options["body_format"]
+    full_case = options["full_case"]
+    reporter_id = options["reporter_id"]
+    source = options["source"]
+    allowed_types = ["jurisdictions", "courts", "reporters", "cases"]
+    base_url = "https://api.case.law"
+    version = "v1"
+    params = {}
+    endpoint = None
+    page_size = 1000
+
+    # TODO test how many cases can we collect, there's api calls limit for registered user?
+    # TODO handle error_auth_required or error_limit_exceeded when want to get casebody
+
+    if not type:
+        logger.error("You didn't provide the type. Exiting.")
+        return
+
+    if type not in allowed_types:
+        logger.error("You didn't provide the type. Exiting.")
+        return
+
+    if last_updated and last_updated_filter:
+        if last_updated_filter not in ["gt", "gte", "lt", "lte"]:
+            logger.error(
+                f"{last_updated_filter} not in options gt, gte, lt, lte."
+            )
+            return
+
+    if not last_updated and last_updated_filter:
+        logger.error(
+            "You can't set last-update-filter without passing last-update argument."
+        )
+        return
+
+    if last_updated and not last_updated_filter:
+        logger.error(
+            "You can't filter by last-update without passing last-update-filter "
+            "argument."
+        )
+        return
```

Comment:
If im not mistaken ... logger.error will send a sentry alert - but if the error is just a ... you used the wrong parameter, we probably just want to throw a logger.warning/warn so as not to clog up sentry